akka-actor/src/main/scala/akka/pattern/FutureTimeoutSupport.scala

```diff
@@ -22,4 +26,29 @@ trait FutureTimeoutSupport {
       using.scheduleOnce(duration) { p completeWith { try value catch { case NonFatal(t) ⇒ Future.failed(t) } } }
       p.future
     }
+
+  /**
+   * Returns a [[scala.concurrent.Future]] that will be completed with the success or failure of the provided value
+   * after the specified duration.
+   */
+  def afterCompletionStage[T](duration: FiniteDuration, using: Scheduler)(value: ⇒ CompletionStage[T])(implicit ec: ExecutionContext): CompletionStage[T] =
```

Comment:
Is this primarily meant to be used by us in the akka sources, not from Java? Asking because of the implicit param.